lib/watcher.ts

```diff
@@ -38,72 +52,49 @@ class Watcher extends EventEmitter {
     if (this.options.debounce == null) {
       this.options.debounce = 100;
     }
-    this._currentBuild = Promise.resolve();
     this.builder = builder;
     this.watcherAdapter =
       this.options.watcherAdapter || new WatcherAdapter(watchedNodes, this.options.saneOptions);
-
-    this._rebuildScheduled = false;
-    this._ready = false;
-    this._quittingPromise = null;
-    this._lifetime = null;
-    this._changedFiles = [];
+    if (this._nextBuild) {
+      this._nextBuild.promise.catch(() => {});
+    }
   }
 
   get currentBuild() {
-    return this._currentBuild;
-  }
-
-  // TODO: this is an interim solution, pending a largely cleanup of this class.
-  // Currently I can rely on understanding the various pieces of this class, to
-  // know this is safe. This is not a good long term solution, but given
-  // relatively little time to address this currently, it is "ok". I do plan,
-  // as time permits to circle back, and do a more thorough refactoring of this
-  // class, to ensure it is safe for future travelers.
-  private _updateCurrentBuild(promise: Promise<void>) {
-    this._currentBuild = promise;
-
-    promise.catch(() => {
-      /**
-       * The watcher internally follows currentBuild, and outputs errors appropriately.
-       * Since watcher.currentBuild is public API, we must allow public follows
-       * to still be informed of rejections.  However we do not want `_currentBuild` itself
-       * to trigger unhandled rejections.
-       *
-       * By catching errors here, but returning `promise` instead of the chain from
-       * `promise.catch`, both goals are accomplished.
-       */
-    });
+    if (this._nextBuild) {
```

Comment:
TODO: toggling between `nextBuild` and `currentBuild` is annoying, lets tidy this up